hydroshare_jupyter_sync/lib/filesystem/fs_resource_map.py

```diff
@@ -0,0 +1,199 @@
+        HydroShare resource's `readme.txt`"""
+        return self.resource_path / self.resource_id
+
+    @property
+    def data_path(self):
+        """Return assumed path to resource data (i.e. `/some/path/{resource_id}/{resource_id}/data/contents`)"""
+        return self.base_directory / "data" / "contents"
+
+    # Helper methods
+    def _abs_path(self, resource_file: Union[Path, str]) -> Path:
+        resource_file = Path(resource_file)
+        return (
+            resource_file
+            if resource_file.is_absolute()
+            else self.base_directory / resource_file
+        ).resolve()
+
+    def _is_member(self, resource_file: Union[Path, str]) -> bool:
+        if self._is_child_of_data_path(resource_file):
+            return self._as_child_of_base_directory(resource_file) in self.data
+        return False
+
+    def _valid_resource_file(self, resource_file: Union[Path, str]) -> bool:
+        abs_path = self._abs_path(resource_file)
+
+        return (
+            abs_path.is_file()
+            and not abs_path.is_symlink()
+            and self._is_child_of_data_path(abs_path)
+        )
+
+    def _as_child_of(
+        self, resource_file: Union[Path, str], parents: Union[Path, str]
+    ) -> Path:
+        abs_path = self._abs_path(resource_file)
+        return abs_path.relative_to(parents)
+
+    def _as_child_of_data_path(self, resource_file: Union[Path, str]) -> Path:
+        return self._as_child_of(resource_file, self.data_path)
+
+    def _as_child_of_base_directory(self, resource_file: Union[Path, str]) -> Path:
+        return self._as_child_of(resource_file, self.base_directory)
+
+    def _is_child_of_data_path(self, resource_file: Union[Path, str]) -> bool:
+        try:
+            # ensure is child of data_path
+            self._as_child_of_data_path(resource_file)
+            return True
+        except ValueError:
+            return False
+
+    def _insert(self, resource_file: Union[Path, str]) -> None:
+        abs_path = self._abs_path(resource_file)
+
+        if self._valid_resource_file(abs_path):
+            # path relative to resource base directory. For comparison, this is how files are listed
+            # in any resource's `manifest-md5.txt` file.
+            truncated_path = abs_path.relative_to(self.base_directory)
+            # compute file md5 hex digest
+            digest = compute_file_md5_hexdigest(abs_path)
+            # insert into collection
+            self.data[truncated_path] = digest
+
+
+class RemoteFSResourceMap(FSResourceMap):
+    def __init__(self, resource: Resource) -> Dict[Path, MD5Hash]:
+        super().__init__()
+        self.resource = resource
+        self.resource_id = resource.resource_id
+
+    @classmethod
+    def from_resource(cls, resource: Resource) -> "FSResourceMap":
+        # create class instance
+        fsresource_map = cls(resource)
+
+        fsresource_map.update_resource()
```

Comment:
This seems like it belongs in the __init__ method as it initializes values.  The classmethod is fine but isn't doing anything the constructor isn't doing besides this method call which initializes values and should be called there too.